tests/test_temperature_processor.py

```diff
@@ -56,7 +59,10 @@ async def test_processor(self) -> None:
 
         timestamp = 12345.0
         response_code = 0
-        sensor_data = [1.0, 2.0, 3.0, 4.0]
+        sensor_data = [1.0, 2.0, 3.0, 4.0, 5.0, None]
+        # Any None value in the sensor data will be replaced with NaN, which is
+        # why the last value in the following list must be set to NaN.
+        expected_sensor_data = [math.nan, 2.0, 3.0, math.nan, 5.0, math.nan]
```

Comment:
Based on conversation with Wouter, this should be np.nan since this is inside of a list.